### PR TITLE
[webtransport] Check existence of Origin header

### PR DESCRIPTION
**Pull request by bashi**
The draft says clients must provide an Origin header [1].

[1] https://datatracker.ietf.org/doc/html/draft-ietf-webtrans-http3-04#section-3.3

### CODE REVIEW

File: tools/webtransport/h3/webtransport_h3_server.py [hunk 1]
```diff
@@ -118,11 +118,13 @@ def _h3_event_received(self, event: H3Event) -> None:
...
             protocol = headers.get(b":protocol")
-            if method == b"CONNECT" and protocol == b"webtransport":
+            origin = headers.get(b"origin")
+            # Accept any Origin but the client must send it.
+            if method == b"CONNECT" and protocol == b"webtransport" and origin:
                 self._session_stream_id = event.stream_id
                 self._handshake_webtransport(event, headers)
             else:
-                self._send_error_response(event.stream_id, 400)
+                self._send_error_response(event.stream_id, 404)
```
Comment (bashi) [inline tools/webtransport/h3/webtransport_h3_server.py:127]: Updated per https://www.ietf.org/archive/id/draft-ietf-webtrans-http3-04.html#section-3.3-3

@vasilvv Which status code a server should use when the server rejects a request due to origin checks?

Comment (vasilvv) [reply tools/webtransport/h3/webtransport_h3_server.py:127]: I believe 403 is the most appropriate error code here.

**Comment by vasilvv** [comment]
Also note that as of ietf-wg-webtrans/draft-ietf-webtrans-http3#94, the Origin header is only required for browser clients (so this probably still makes sense for WPT).

File: tools/webtransport/h3/webtransport_h3_server.py (hunk 1 above)
Comment (bashi) [reply tools/webtransport/h3/webtransport_h3_server.py]: Updated to use 403 for origin check failure. Thanks!

**Comment by bashi** [comment]
Uploaded PR to clarify origin header checks in the draft.
https://github.com/ietf-wg-webtrans/draft-ietf-webtrans-http3/pull/98

**Review by nidhijaju — APPROVED** [review]
LGTM, thank you for updating this!

**Pull request merged**